### PR TITLE
Toggle box

### LINKED ISSUE #531: igx-toggle implementation
### Description  
Implement igx-toggle directive

### PR DESCRIPTION
**Pull request by Aleksandyr**
Closes #531

### CODE REVIEW

File: src/directives/toggle/toggle.directive.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    @Input()
+    public id: string;
+
+    public get element() {
+        return this.elementRef.nativeElement;
+    }
+
+    @HostBinding("class.igx-toggle--hidden")
+    protected get hiddenClass() {
```
Comment (rkaraivanov) [inline src/directives/toggle/toggle.directive.ts]: This should be `public`

File: src/directives/toggle/toggle.directive.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        return this.collapsed;
+    }
+
+    @HostBinding("class.igx-toggle")
+    protected get defaultClass() {
```
Comment (rkaraivanov) [inline src/directives/toggle/toggle.directive.ts]: This should be `public`

File: src/directives/toggle/toggle.directive.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                this.target.close(true);
+                document.removeEventListener("click", this._handler, true);
+            };
+
+            document.addEventListener("click", this._handler, true);
+        }
+    }
+
+    @HostListener("click", ["$event"])
+    private onClick() {
```
Comment (rkaraivanov) [inline src/directives/toggle/toggle.directive.ts]: Again this should be `public`.

File: src/directives/toggle/toggle.directive.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+                this.target.close(true);
+                document.removeEventListener("click", this._handler, true);
+            };
+
+            document.addEventListener("click", this._handler, true);
+        }
+    }
+
+    @HostListener("click", ["$event"])
```
Comment (rkaraivanov) [inline src/directives/toggle/toggle.directive.ts]: The `$event` passed to the decorator is redundant as it is not passed on the handler itself. You can remove it.

File: src/directives/toggle/toggle.directive.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    }
+}
+
+@Directive({
+    exportAs: "toggle-action",
+    selector: "[igxToggleAction]"
+})
+export class IgxToggleActionDirective implements OnDestroy, OnInit {
+    @Input()
+    public closeOnOutsideClick = false;
```
Comment (rkaraivanov) [inline src/directives/toggle/toggle.directive.ts]: I think the default expectation is to close the content marked with the directive when clicking outside of it. Did not see anything mentioned about it in the spec but I feel this should be `true` by default. 
@kdinev what is your take on this?

File: README.md [hunk 6]
```diff
@@ -42,6 +42,7 @@ Current list of controls include:
 | **switch**            | Available | [Readme](https://github.com/IgniteUI/igniteui-angular/blob/master/src/switch/README.md)                        |     |               |               |                                                                                                    |
 | **tabbar**            | Available | [Readme](https://github.com/IgniteUI/igniteui-angular/blob/master/src/tabbar/README.md)                        |     |               |               |                                                                                                    |
 | **toast**             | Available | [Readme](https://github.com/IgniteUI/igniteui-angular/blob/master/src/toast/README.md)                         |     |               |               |                                                                                                    |
+| **toast**             | Available | [Readme](https://github.com/IgniteUI/igniteui-angular/blob/master/src/directives/README-TOGGLE.md)                         |     |               |               |                                                                                                    |
```
Comment (kdinev) [inline README.md]: This should be **toggle** and not **toast**

**Pull request merged**